### PR TITLE
Link project to Azure Key Vault

### PR DESCRIPTION
**Pull request by gregzealley**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,13 +4,22 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
+    <parent>
```
Comment (JoelClemence) [inline pom.xml]: Unless you really need this I would suggest using the Spring Maven BOM as per: https://www.baeldung.com/spring-maven-bom (Like you've done with the Microsoft dependencies)

File: pom.xml [hunk 2]
```diff
@@ -4,13 +4,22 @@
...
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.1.7.RELEASE</version>
```
Comment (JoelClemence) [inline pom.xml]: Easier to manage if this is a property and referenced like the other versions like we did in MINT

File: src/main/java/uk/gov/ukho/apitest/ApiFrameworkDemoApplication.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class ApiFrameworkDemoApplication {
+    public static void main(String[] args) {
+        SpringApplication.run(ApiFrameworkDemoApplication.class, args);
+    }
+}
```
Comment (JoelClemence) [inline src/main/java/uk/gov/ukho/apitest/ApiFrameworkDemoApplication.java]: New line at end of file

File: pom.xml [hunk 4]
```diff
@@ -47,7 +77,29 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+
 </project>
```
Comment (JoelClemence) [inline pom.xml]: new line at end of file

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,3 @@
+azure.keyvault.uri=<url of Azure Vault>
+azure.keyvault.client-id=<guid of Azure client ID>
+azure.keyvault.client-key=<guid of Azure client password>
```
Comment (JoelClemence) [inline src/main/resources/application.properties]: New line

File: src/test/java/uk/gov/ukho/apitest/services/ExampleJsonApiService.java [hunk 6]
```diff
@@ -10,8 +10,11 @@
 public class ExampleJsonApiService {
 
     private RequestSpecification spec;
+    private String apiBaseEndpoint;
 
-    private final String API_BASE = "https://jsonplaceholder.typicode.com";
+    public ExampleJsonApiService(String apiBaseEndpoint) {
```
Comment (JoelClemence) [inline src/test/java/uk/gov/ukho/apitest/services/ExampleJsonApiService.java]: Final argument?

File: src/test/java/uk/gov/ukho/apitest/services/ExampleJsonApiService.java [hunk 7]
```diff
@@ -10,8 +10,11 @@
 public class ExampleJsonApiService {
 
     private RequestSpecification spec;
```
Comment (JoelClemence) [inline src/test/java/uk/gov/ukho/apitest/services/ExampleJsonApiService.java:12]: Can  these variables be final?

**Review by JoelClemence — CHANGES_REQUESTED** [review]
Only small comments

File: pom.xml (hunk 2 above)
Comment (gregzealley) [reply pom.xml]: IntelliJ says no - not possible to use properties here

File: src/main/java/uk/gov/ukho/apitest/ApiFrameworkDemoApplication.java (hunk 3 above)
Comment (gregzealley) [reply src/main/java/uk/gov/ukho/apitest/ApiFrameworkDemoApplication.java]: Done

File: pom.xml (hunk 4 above)
Comment (gregzealley) [reply pom.xml]: Done

File: src/main/resources/application.properties (hunk 5 above)
Comment (gregzealley) [reply src/main/resources/application.properties]: Done

File: src/test/java/uk/gov/ukho/apitest/services/ExampleJsonApiService.java (hunk 6 above)
Comment (gregzealley) [reply src/test/java/uk/gov/ukho/apitest/services/ExampleJsonApiService.java]: Done

File: src/test/java/uk/gov/ukho/apitest/services/ExampleJsonApiService.java (hunk 7 above)
Comment (gregzealley) [reply src/test/java/uk/gov/ukho/apitest/services/ExampleJsonApiService.java:12]: One of them can...

**Review by JoelClemence — APPROVED** [review]

**Pull request merged**